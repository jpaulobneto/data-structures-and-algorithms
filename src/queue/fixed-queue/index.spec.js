const FixedQueue = require('.');

describe('FixedQueue', () => {
  it('is defined', () => {
    expect(FixedQueue).toBeDefined();
  });

  it('can be initialized', () => {
    expect(() => new FixedQueue()).not.toThrow();
  });

  describe('isEmpty()', () => {
    it('return true if queue is empty', () => {
      const queue = new FixedQueue();

      expect(queue.isEmpty()).toEqual(true);
    });
  });

  describe('isFull()', () => {
    it('return false if queue is not full', () => {
      const queue = new FixedQueue();

      expect(queue.isFull()).toEqual(false);
    });
  });

  describe('enqueue(value)', () => {
    it('10x queue initial size', () => {
      const queue = new FixedQueue();
      const values = [];

      for (let i = 1; i <= 20; i += 1) {
        values.push(i);
      }

      values.forEach(value => queue.enqueue(value));

      expect(queue).toMatchSnapshot();
    });
  });

  describe('dequeue()', () => {
    it('remove the next value from queue', () => {
      const queue = new FixedQueue();
      const values = [];

      for (let i = 1; i <= 20; i += 1) {
        values.push(i);
      }

      values.forEach((value) => {
        if (value % 10 === 0) queue.dequeue();

        queue.enqueue(value);
      });

      expect(queue).toMatchSnapshot();
    });
  });
});
