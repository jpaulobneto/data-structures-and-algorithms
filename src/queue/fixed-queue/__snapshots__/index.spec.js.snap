// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FixedQueue dequeue() remove the next value from queue 1`] = `
Queue {
  "head": 1,
  "size": 18,
  "storage": Array [
    20,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
  ],
  "tail": 1,
}
`;

exports[`FixedQueue enqueue(value) 10x queue initial size 1`] = `
Queue {
  "head": 0,
  "size": 20,
  "storage": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    12,
    13,
    14,
    15,
    16,
    17,
    18,
    19,
    20,
  ],
  "tail": 0,
}
`;
